src/go/rpk/pkg/cli/debug/bundle/bundle_k8s_linux.go

```diff
@@ -541,12 +583,22 @@ func saveK8SLogs(ctx context.Context, ps *stepParams, namespace, since string, l
 
 		var grp multierror.Group
 		for _, p := range pods.Items {
-			p := p
-			cb := func(ctx context.Context) ([]byte, error) {
-				return podsInterface.GetLogs(p.Name, logOpts).Do(ctx).Raw()
+			for _, c := range p.Spec.Containers {
+				opts := logOpts.DeepCopy()
+				opts.Container = c.Name
+				cb := func(ctx context.Context) ([]byte, error) {
+					return podsInterface.GetLogs(p.Name, opts).Do(ctx).Raw()
+				}
+				grp.Go(func() error { return requestAndSave(ctx, ps, fmt.Sprintf("logs/%v-%v.txt", p.Name, c.Name), cb) })
```

Comment:
I _think_ this is fine as of go 1.23 but the reference to `p` and `c` in a closure from another goroutine inside a doublely nested loop has all of my internal go alarms ringing.

Just to check have you tested this on a cluster with multiple containers and Pods and observed that the logs are all named correctly?

Reply:
Correct, I double checked just now and the logs are named and saved correctly 
![image](https://github.com/user-attachments/assets/38301bc2-d3db-40f6-a447-7a2bc841b172)